tst/SignalingApiTest.cpp

```diff
@@ -9,11 +9,57 @@ namespace webrtcclient {
 class SignalingApiTest : public WebRtcClientTestBase {
 };
 
+TEST_F(SignalingApiTest, invalidRoleType_deserializeSignalingCacheEntries)
+{
+    FREMOVE(DEFAULT_CACHE_FILE_PATH);
+    srand(GETTIME());
+
+    BOOL cacheFound = FALSE;
+    SignalingFileCacheEntry testEntry;
+    MEMSET(&testEntry, 0x00, SIZEOF(testEntry));
+    UINT32 serializedCacheEntryLen;
+    CHAR testWssEndpoint[MAX_SIGNALING_ENDPOINT_URI_LEN + 1] = {0};
+    CHAR testHttpsEndpoint[MAX_SIGNALING_ENDPOINT_URI_LEN + 1] = {0};
+    CHAR testWebrtcEndpoint[MAX_SIGNALING_ENDPOINT_URI_LEN + 1] = {0};
+    CHAR testRegion[MAX_REGION_NAME_LEN + 1] = {0};
+    CHAR testChannelArn[MAX_ARN_LEN + 1] = {0};
+    CHAR testChannel[MAX_CHANNEL_NAME_LEN + 1] = {0};
+    UINT64 time = GETTIME() / HUNDREDS_OF_NANOS_IN_A_SECOND;
+    int append = 0;
+    CHAR serializedCacheEntry[MAX_SERIALIZED_SIGNALING_CACHE_ENTRY_LEN];
+
+
+    const int TEST_CHANNEL_COUNT = 5;
+
+    MEMSET(testWssEndpoint, 0, MAX_SIGNALING_ENDPOINT_URI_LEN+1);
+    MEMSET(testHttpsEndpoint, 0, MAX_SIGNALING_ENDPOINT_URI_LEN+1);
+    MEMSET(testRegion, 0, MAX_REGION_NAME_LEN + 1);
+    MEMSET(testChannelArn, 0, MAX_ARN_LEN+1);
+    MEMSET(testChannel, 0, MAX_CHANNEL_NAME_LEN+1);
+
+    append = rand() % TEST_CHANNEL_COUNT;
+    SNPRINTF(testWssEndpoint, SIZEOF(testWssEndpoint), "%s%d", "testWssEndpoint", append);
+    SNPRINTF(testHttpsEndpoint, SIZEOF(testHttpsEndpoint),"%s%d", "testHttpsEndpoint", append);
+    SNPRINTF(testRegion, SIZEOF(testRegion),"%s%d", "testRegion", append);
+    SNPRINTF(testChannelArn, SIZEOF(testChannelArn),"%s%d", "testChannelArn", append);
+    SNPRINTF(testChannel, SIZEOF(testChannel),"%s%d", "testChannel", append);
+
+    EXPECT_EQ(STATUS_SUCCESS, createFileIfNotExist(DEFAULT_CACHE_FILE_PATH));
```

Comment:
Do we also need to clear it just in case it exists from a previous test or something

Reply:
Done.

Reply:
The file removal already exists in the start of the function. And I am directly writing the entry to the file.